src/status_im/contexts/wallet/events.cljs

```diff
@@ -69,7 +69,16 @@
 
 (rf/reg-event-fx :wallet/switch-current-viewing-account
  (fn [{:keys [db]} [address]]
-   {:db (assoc-in db [:wallet :current-viewing-account-address] address)}))
+   {:db (-> db
+            (assoc-in [:wallet :current-viewing-account-address] address)
+            (update-in [:wallet :ui :swap]
+                       dissoc
+                       :last-request-uuid
+                       :swap-proposal
+                       :error-response
+                       :loading-swap-proposal?
+                       :approval-transaction-id
+                       :approved-amount))}))
```

Comment:
it's a bit strange that we have "swap specific" functionality in the `:wallet/switch-current-viewing-account` event. Might be better to reset these fields in a declarative way in the input component's `use-effect`, since we already are using it to re-fetch the proposal.

Reply:
It is weird, actually. I'll come up with a different approach and let you know.

Reply:
Updated!